extends Area2D

@export var skilltext: Label              # æ‹–å…¥é¡¯ç¤ºç­‰ç´šçš„ Label
@export var costtext: Label               # æ‹–å…¥é¡¯ç¤ºé‡‘é¡çš„ Label
@export var skill_index: int = 1          # é€™å€‹æŠ€èƒ½æ˜¯ç¬¬å¹¾å€‹ï¼ˆ1ï½4ï¼‰
@export var totalframes: int = 2          # å‡ç´šæ¬¡æ•¸ï¼ˆ2 æœ‰ 3 å€‹éšæ®µï¼‰
@export var skill_sprite: Sprite2D 

var current_frame := 0

var upgrade_costs = {
	1: [25, 75],
	2: [50, 150],
	3: [100, 300],
	4: [50, 150]
}

func _ready():
	update_texts()

func _input_event(_viewport, event, _shape_idx):
	if event is InputEventMouseButton and event.pressed:
		if current_frame < totalframes:
			var cost = upgrade_costs[skill_index][current_frame]
			if GlobalProperty.Property["Coin"] >= cost:
				GlobalProperty.Property["Coin"] -= cost

				match skill_index:
					1:
						skillStats.skill1Upgrade()
						print("ğŸ”º Skill1 å‡ç´šï¼šDamage = ", skillStats.Skill1DamageGetter(), ", CD = ", skillStats.skill1CDGetter())
					2:
						skillStats.specialUpgrade() ## remember change
					3:
						skillStats.ULTUpgrade()
					4:
						skillStats.healthUpgrade()
				current_frame += 1
				skill_sprite.frame = current_frame
				skillStats.set_skill_level(skill_index)
				update_texts()
			else:
				print("âŒ é‡‘å¹£ä¸è¶³ï¼Œéœ€ï¼š", cost)
		else:
			print("âœ… å·²é”æœ€é«˜ç­‰ç´š")

func update_texts():
	skilltext.text = "Lv " + str(current_frame+1)
	if current_frame < totalframes:
		var next_cost = upgrade_costs[skill_index][current_frame]
		costtext.text = str(next_cost)
	else:
		costtext.text = "MAX"
